spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/oauth2/resource/servlet/OAuth2ResourceServerJwtConfiguration.java

```diff
@@ -79,8 +91,19 @@ public JwtDecoder jwtDecoderByPublicKeyValue() throws Exception {
 
 		@Bean
 		@Conditional(IssuerUriCondition.class)
-		public JwtDecoder jwtDecoderByIssuerUri() {
-			return JwtDecoders.fromOidcIssuerLocation(this.properties.getIssuerUri());
+		public JwtDecoder jwtDecoderByIssuerUri(ObjectProvider<OAuth2TokenValidator<Jwt>> jwtValidators) {
+			NimbusJwtDecoder jwtDecoder = (NimbusJwtDecoder) JwtDecoders
```

Comment:
I'm not sure about the unchecked class cast here. If something were to change in Spring Security where it no longer returned a `NimbusJwtDecoder`, this would break.

Reply:
Yes, and same for reactive one as well.
It can add `instanceof` check.
What is the behavior if it does not match the `instanceof` check?
Is it ok to silently ignore, or should it fail?

Reply:
Failing would potentially be a breaking change if the `JwtDecoder` was no longer a `NimbusJwtDecoder`. Let's see what the rest of the team thinks.